backend/src/main/java/ai/verta/modeldb/dataset/DatasetDAORdbImpl.java

```diff
@@ -93,63 +93,63 @@ public DatasetDAORdbImpl(AuthService authService, RoleService roleService) {
     this.roleService = roleService;
   }
 
-  private void checkDatasetAlreadyExist(Dataset dataset) {
-    try (Session session = ModelDBHibernateUtil.getSessionFactory().openSession()) {
-      ModelDBHibernateUtil.checkIfEntityAlreadyExists(
-          session,
-          "ds",
-          CHECK_DATASE_QUERY_PREFIX,
-          "Dataset",
-          "datasetName",
-          dataset.getName(),
-          ModelDBConstants.WORKSPACE,
-          dataset.getWorkspaceId(),
-          dataset.getWorkspaceType(),
-          LOGGER);
-    }
+  private void checkDatasetAlreadyExist(Session session, Dataset dataset) {
+    ModelDBHibernateUtil.checkIfEntityAlreadyExists(
+        session,
+        "ds",
+        CHECK_DATASE_QUERY_PREFIX,
+        "Dataset",
+        "datasetName",
+        dataset.getName(),
+        ModelDBConstants.WORKSPACE,
+        dataset.getWorkspaceId(),
+        dataset.getWorkspaceType(),
+        LOGGER);
   }
 
   @Override
   public Dataset createDataset(Dataset dataset, UserInfo userInfo)
       throws InvalidProtocolBufferException {
-    // Check entity already exists
-    checkDatasetAlreadyExist(dataset);
+    createRolesForDataset(dataset, userInfo);
```

Comment:
What happens if it already exists? Will the role service return an error? Or maybe add duplicated entries? I'd recommend still performing the check first. You can use a separate session for that, right?

Reply:
this is moving into the below single transaction. Role service update do nothing if role binding is already added but I will confirm it again.

Reply:
Please do and let me know here.

Reply:
It is created a new role binding with a new dataset id so I am reverting my change and add exists check before the role creation.